docs/book/developer-guide/advanced-concepts/docker.md

```diff
@@ -0,0 +1,75 @@
+---
+description: How ZenML uses Docker images to run your pipeline
+---
+
```

Comment:
I feel like this needs more easier to read motivation. Something like "When running locally, ZenML does not use docker, but when a container registry and remote orchestrator/step operator is in your stack it behaves differnetly. Now it needs to transport your code and config to a remote location. ZenML achieves this by..."